src/test/java/seedu/address/logic/commands/AddAllocCommandTest.java

```diff
@@ -48,12 +48,16 @@ public void execute_addAllocByName_success() {
         AddAllocCommand addAllocCommand = new AddAllocCommand(allocDescriptor);
 
         model.addStudent(simpleAmy);
+
+        Model expectedModel = new ModelManager(new AddressBook(model.getAddressBook()), new UserPrefs());
+
+        Group simpleGroupCopy = new Group(simpleGroup.name);
         model.addGroup(simpleGroup);
+        expectedModel.addGroup(simpleGroupCopy);
 
         Student expectedAmy = new PersonBuilder(simpleAmy)
                 .withGroups(VALID_GROUP_TUTORIAL).build();
 
-        Model expectedModel = new ModelManager(new AddressBook(model.getAddressBook()), new UserPrefs());
         expectedModel.setStudent(simpleAmy, expectedAmy);
```

Comment:
Tests are changed here. A copy of the group is created, if not when we do expectedModel.setStudent(simpleAmy, expectedAmy);, it adds the student to the group and the command cannot be executed since the student already exists in the group.

Reply:
Yeah this makes sense.